lib/floe/workflow/choice_rule/data.rb

```diff
@@ -4,6 +4,24 @@ module Floe
   class Workflow
     class ChoiceRule
       class Data < Floe::Workflow::ChoiceRule
+        attr_reader :compare_key
+
+        COMPARE_KEYS = (
+          %w[IsNull IsPresent IsNumeric IsString IsBoolean IsTimestamp StringMatches] +
+          %w[String Numeric Boolean Timestamp].flat_map { |k| ["#{k}Equals", "#{k}EqualsPath"] } +
+          %w[String Numeric Timestamp].flat_map { |k| %w[LessThan GreaterThan LessThanEquals GreaterThanEquals].flat_map { |x| ["#{k}#{x}", "#{k}#{x}Path"] } }
+        ).freeze
```

Comment:
would have preferred using a hash mapping the string to the operation.
Then we could drop the switch

But this is good for now

Reply:
That's a good idea let me give that a shot